File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        
        vector<int> ans(2);
        unordered_map<int, int> map;
        for(int i=0; i<nums.size(); i++)
        {
            int newTarget = target - nums[i];
            if(map.find(newTarget)!=map.end())
            {
                ans[0] = i;
                ans[1] = map[newTarget];
            }
            map.insert({nums[i], i});
        }
      
        return ans;   
    }
};

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        
        int [] ans = new int[2];
        HashMap<Integer, Integer> hs = new HashMap<>();
        
        for(int i=0; i<nums.length; i++)
        {
            int newTarget = target - nums[i];
            if(hs.containsKey(newTarget))
            {
                ans[0] = i;
                ans[1] = hs.get(newTarget);
            }
            hs.put(nums[i], i);
        }
        return ans;
        
    }
}